src/utils/details.js

```diff
@@ -15,6 +15,7 @@ function _createRequest ({
   headers,
   data
 }) {
+  validateHeaders(headers)
   const statusLineText = statusLine ? 'PSK/1.0\n' : ''
   const headerLines = Object.keys(headers)
     .map((k) => k + ': ' + headers[k])
```

Comment:
Maybe move this into the validateHeaders function, so it only loops over the object once.